crates/wasm-compose/src/graph.rs

```diff
@@ -438,6 +472,86 @@ pub struct EncodeOptions {
     pub validate: bool,
 }
 
+#[derive(Clone, Debug, Default)]
+pub(crate) struct ResourceMapping {
+    map: im_rc::HashMap<ResourceId, (ComponentId, ResourceId)>,
+}
+
+impl ResourceMapping {
+    pub(crate) fn add_pairs(
+        mut self,
+        export_component: ComponentId,
+        export_type: ComponentEntityType,
+        export_types: TypesRef,
+        _import_component: ComponentId,
+        import_type: ComponentEntityType,
+        import_types: TypesRef,
+    ) -> Option<Self> {
+        if let (
+            ComponentEntityType::Instance(export_type),
+            ComponentEntityType::Instance(import_type),
+        ) = (export_type, import_type)
+        {
+            let mut exports = HashMap::new();
+            for (export_name, ty) in &export_types[export_type].exports {
+                // TODO: support nested instances
```

Comment:
How difficult do you think it would be to solve this TODO and the ones below which derive from a similar code structure?

I was thinking there could be something along the lines of "invoke this callback argument on all resources found" or similar?

Reply:
Not difficult, but I'm lazy so I've opened an issue instead: https://github.com/bytecodealliance/wasm-tools/issues/1288